### ISSUE TITLE
Example on the dune home page does not work

### ISSUE THREAD

**Issue by yawaramin** [ocaml/dune#7457]
## Expected Behavior

Try out the minimal example on https://dune.build/

The given example and shell commands run successfully.

## Actual Behavior

Get an error.

## Reproduction

Equivalent repro:

```
$ dune --version
3.6.1
$ mkdir dune_eg
$ cd dune_eg 
$ echo '(executable (name dune_eg))' >dune
$ echo 'let () = print_endline "hello"' >dune_eg.ml
$ dune build dune_eg.ml
Error: I cannot find the root of the current workspace/project.
If you would like to create a new dune project, you can type:

    dune init project NAME

Otherwise, please make sure to run dune inside an existing project or
workspace. For more information about how dune identifies the root of the
current workspace/project, please refer to
https://dune.readthedocs.io/en/stable/usage.html#finding-the-root
```

**Comment by yawaramin** [comment]
Wouldn't it be simpler for users if dune would just create the `dune-project` file for us? It knows that it needs just the line `(lang dune 3.6)` (or whatever version), the occasional human dune user will likely not remember that exact syntax. Following the suggestion of `dune init project NAME` creates the project `NAME` as a subdirectory _inside_ the existing project, which is not really a great workflow.

**Comment by emillon** [comment]
Let's keep the 2 discussions separate:
- the example on the website can indeed be expanded. can you open a PR on https://github.com/ocaml/dune-www and edit `index.md` accordingly?
- we can revisit the decision wheter to generate `dune-project` files for dune 4. This will happen in a different issue.

**Comment by yawaramin** [comment]
@Alizter dune is of course backwards compatible but it's not forward-compatible 😉 if you have `(lang dune 3.6)` then you can't use dune 3.7 features. Right?

**Comment by Alizter** [comment]
Right, but it is very unlikely that we would put Dune 3.7 features in that example.

**Comment by yawaramin** [comment]
In the minimal getting started example, of course. But what will happen is people use that to bootstrap a dune project, then later they try to upgrade dune and use 3.7+ features in the project and run into an error.

**Comment by Alizter** [comment]
@yawaramin When you use a 3.7 feature but your dune-project file has an older version Dune will give you an error message saying that the feature is only available in 3.7 and that you should update the version in your dune-project file.

**Comment by yawaramin** [comment]
OK, fair enough. In that case my PR should cover it.

**Comment by emillon** [comment]
Yes, this example does not need to be "kept up to date", it probably works with lang 1.0.
